src/ol/uri.js

```diff
@@ -0,0 +1,17 @@
+goog.provide('ol.uri');
+
+
+/**
+ * Appends query parameters to a URI.
+ *
+ * @param {string} uri The original URI, which may already have query data.
+ * @param {!Object} params An object where keys are URI-encoded parameter keys,
+ *     and the values are arbitrary types or arrays.
+ * @return {string} The new URI.
+ */
+ol.uri.appendParams = function(uri, params) {
+  var qs = Object.keys(params).map(function(k) {
+    return encodeURIComponent(k) + '=' + encodeURIComponent(params[k]);
+  });
+  return uri + (uri.indexOf('?') === -1 ? '?' : '&') + qs.join('&');
```

Comment:
just tried some tests on these examples. In neither case does it seem to matter if there is a duplicate '??' or '&&'. However, the current `goog.uri` function removes any duplicates and if there are browsers/servers where this would create an error, it would be a regression if this new function doesn't do this. It should be pretty easy to remove the duplicate by not adding the '?' or '&' if `uri.slice(-1)` is '?' or '&' (or alternatively `uri = uri.slice(0,-1)`) or whatever.